src/Npgsql/NpgsqlConnector.cs

```diff
@@ -584,7 +584,7 @@ async Task RawOpen(NpgsqlTimeout timeout, bool async, CancellationToken cancella
                     RelaxedTextEncoding = Encoding.GetEncoding(Settings.Encoding, EncoderFallback.ReplacementFallback, DecoderFallback.ReplacementFallback);
                 }
 
-                ReadBuffer = new NpgsqlReadBuffer(this, _stream, _socket, Settings.ReadBufferSize, TextEncoding, RelaxedTextEncoding);
+                ReadBuffer = new NpgsqlReadBuffer(this, _stream, _socket, Settings.ReadBufferSize, !Settings.Pooling, TextEncoding, RelaxedTextEncoding);
```

Comment:
Any reason for the array pooling feature to depend on the connection pooling setting in the settings? Any reason to not just always turn it on?

Reply:
Because long living connections may take more memory from the pool than requested and not fully utilize it.